custom/abt/reports/late_pmt.py

```diff
@@ -204,17 +204,15 @@ def _to_report_format(date, user, group):
         rows = []
         sub_status = self.report_config['submission_status']
         if users:
-            group_a = self.get_users_in_group_a
-            group_b = self.get_users_in_group_b
-
             for date in dates:
                 for user in users:
-                    key = (date.date(), user['user_id'])
-                    if key not in group_a and sub_status != 'group_b':
-                        group = _('No PMT data Submitted')
-                    elif key not in group_b and key in group_a and sub_status != 'group_a':
-                        group = _('Incorrect PMT data Submitted')
+                    sms_received = (date.date(), user['user_id']) in self.smss_received
+                    sms_event_exists = (date.date(), user['user_id']) in self.sms_events
+                    if not sms_received and sub_status != 'group_b':
+                        error_msg = _('No PMT data Submitted')
+                    elif not sms_event_exists and sms_received and sub_status != 'group_a':
```

Comment:
It might improve readability to create constants like
```python
SMS_RECEIVED_FROM_MW = 'group_a'
SMS_SENT_TO_MW = 'group_b'
```
and then to use those instead of the values "group_a" and "group_b", whose meanings aren't immediately obvious. (And if I've misunderstood their meanings here, obviously please rename. :upside_down_face: )

Reply:
Agreed - I think it could be done higher up this function, like:
```python
include_missing = sub_status != 'group_b'
include_incorrect = sub_status != 'group_a'
```